src/js/service/ImportService.js

```diff
@@ -80,6 +80,73 @@
   };
 
   /**
+   * Given an image object and some options, merge the frames with the currently open
+   * Piskel and open it for editing.
+   * @param {!Image} image
+   * @param {!Object} options
+   * @param {!string} options.importType - 'single' if not spritesheet
+   * @param {!number} options.frameSizeX
+   * @param {!number} options.frameSizeY
+   * @param {number} [options.frameOffsetX] only used in spritesheet imports.
+   * @param {number} [options.frameOffsetY] only used in spritesheet imports.
+   * @param {!boolean} options.smoothing
+   * @param {number} [options.frameRate] in frames per second
+   * @param {function} [onComplete]
+   */
+  ns.ImportService.prototype.importFramesFromImage = function (image, options, onComplete) {
+    onComplete = onComplete || Constants.EMPTY_FUNCTION;
```

Comment:
ImportService.js [was added by us](https://github.com/juliandescottes/piskel/commits/master/src/js/service/ImportService.js) - refactor away.  I'd strongly consider submitting a generic version of this change upstream too.